src/olympia/migrations/1026-add-position-column-to-version-preview.sql

```diff
@@ -0,0 +1,5 @@
+ALTER TABLE `version_previews`
+    ADD COLUMN position TINYINT(11) unsigned DEFAULT 0;
+
+CREATE INDEX version_position_created_idx
+    ON version_previews (`version_id`, `position`, `created`);
```

Comment:
It was supposed to match the ordering I added (` ordering = ('position', 'created')`) to `VersionPreview` model below.  My thinking was it would always be filtered by version.  That said, my understanding of sql indexes is limited at best.  Should it be just ```CREATE INDEX position_created_idx ON version_previews (`position`, `created`)```?

Reply:
You can test this using `EXPLAIN EXTENDED …sql query here…` which should give you more details about whether this is being used. For that, you'll have to log the database queries though (either via `print(qs.query)`, django-debug-toolbar or similar tools)

IMHO, `(position, created)` should suffice, you can however also test `(position, created, version_id)` in case the version id get's regularly queried *together* with position and created (which I kinda doubt). Indexes are evaluated from left to right, e.g mysql uses either the leftmost column, or the two leftmost columns, or the three leftmost columns etc.

Reply:
Why don't you want `version_id` in there? The most used query is definitely going to have all 3 columns, as it's coming from `self.current_version.previews.all` (where `self` is an `Addon`). With the default ordering being what it is in this PR, the index @eviljeff created makes sense to me.

Reply:
You can add an `index_together` to be nice with django migrations in the future though :) (bonus point if the index name matches the name django would have used)

Reply:
I put `print self.current_version.previews.all().query` in `Addon.current_previews`:
```sql
SELECT `version_previews`.`id`, `version_previews`.`created`, `version_previews`.`modified`, `version_previews`.`version_id`, `version_previews`.`position`, `version_previews`.`sizes`
FROM `version_previews` WHERE `version_previews`.`version_id` = 129
ORDER BY `version_previews`.`position` ASC, `version_previews`.`created` ASC
```
using `EXPLAIN EXTENDED`:
```sql
MySQL [olympia]> explain extended SELECT `version_previews`.`id`, `version_previews`.`created`, `version_previews`.`modified`, `version_previews`.`version_id`, `version_previews`.`position`, `version_previews`.`sizes` FROM `version_previews` WHERE `version_previews`.`version_id` = 129 ORDER BY `version_previews`.`position` ASC, `version_previews`.`created` ASC;
+----+-------------+------------------+------------+------+-------------------------------------------------------------------------+------------------------------+---------+-------+------+----------+-----------------------+
| id | select_type | table            | partitions | type | possible_keys                                                           | key                          | key_len | ref   | rows | filtered | Extra                 |
+----+-------------+------------------+------------+------+-------------------------------------------------------------------------+------------------------------+---------+-------+------+----------+-----------------------+
|  1 | SIMPLE      | version_previews | NULL       | ref  | version_previews_version_id_fk_versions_id,version_position_created_idx | version_position_created_idx | 4       | const |    1 |   100.00 | Using index condition |
+----+-------------+------------------+------------+------+-------------------------------------------------------------------------+------------------------------+---------+-------+------+----------+-----------------------+
```

I'm reading that as the query `ORDER BY` doesn't reference `version_id` but the index (including `version_id`) _was_ used for the query.

Reply:
It does use `version_id` in the `WHERE`, as expected. So the index works perfectly. You're good.